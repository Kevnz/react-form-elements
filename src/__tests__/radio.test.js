import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import Radio from '../radio'

const LABEL_TEXT = 'My Checkbox Label'
describe('Radio component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(<Radio label={LABEL_TEXT} />)

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText(LABEL_TEXT)
      expect(label.textContent).toBe(LABEL_TEXT)
    })
    it('should not be checked', () => {
      const { getByLabelText, container } = render(
        <Radio label={LABEL_TEXT} initialValue="The Value" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const check = getByLabelText(LABEL_TEXT)
      expect(check).toMatchSnapshot()
      expect(check.checked).toBe(false)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const { getByLabelText, container } = render(
        <Radio label={LABEL_TEXT} initialValue="The Value" checked />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText(LABEL_TEXT)
      expect(input.checked).toBe(true)
    })
    it('should update a value when changed in', () => {
      const { getByLabelText, container } = render(
        <Radio label={LABEL_TEXT} initialValue="The Value" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText(LABEL_TEXT)
      fireEvent.change(input, { target: { value: 43 } })
      expect(input.value).toBe('43')
    })
  })
})
