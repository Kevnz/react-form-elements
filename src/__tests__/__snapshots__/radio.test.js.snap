// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My Checkbox Label
  </label>
  <input
    id="id-1-1"
    type="radio"
    value="myradio"
  />
</div>
`;

exports[`Radio component handling input should set a value when passed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Label
  </label>
  <input
    checked=""
    id="id-2-1"
    type="radio"
    value="myradio"
  />
</div>
`;

exports[`Radio component handling input should update a value when changed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-3-1"
  >
    My Label
  </label>
  <input
    id="id-3-1"
    type="radio"
    value="myradio"
  />
</div>
`;
